src/Form/AnswersHandler/AnswersHandler.php

```diff
@@ -0,0 +1,101 @@
+<?php
+
+/**
+ * ---------------------------------------------------------------------
+ *
+ * GLPI - Gestionnaire Libre de Parc Informatique
+ *
+ * http://glpi-project.org
+ *
+ * @copyright 2015-2024 Teclib' and contributors.
+ * @copyright 2003-2014 by the INDEPNET Development Team.
+ * @licence   https://www.gnu.org/licenses/gpl-3.0.html
+ *
+ * ---------------------------------------------------------------------
+ *
+ * LICENSE
+ *
+ * This file is part of GLPI.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ *
+ * ---------------------------------------------------------------------
+ */
+
+namespace Glpi\Form\AnswersHandler;
+
+use Glpi\Form\AnswersSet;
+use Glpi\Form\Form;
+use Session;
+
+/**
+ * Helper class to handle raw answers data
+ * TODO: singleton
+ */
+class AnswersHandler
+{
+    /**
+     * Saves the given answers of a given form into an AnswersSet object
+     *
+     * @param Form  $form    The form to save answers for
+     * @param array $answers The answers to save
+     *
+     * @return AnswersSet|false The created AnswersSet object or false on failure
+     */
+    public function saveAnswers(Form $form, array $answers): AnswersSet|false
+    {
+        /** @var \DBmysql $DB */
+        global $DB;
+
+        // Find next answer index for this form
+        $next_index = 1;
+        $rows = $DB->request([
+            'SELECT' => ['MAX' => 'index AS current_index'],
+            'FROM'   => AnswersSet::getTable(),
+            'WHERE'  => [
+                'forms_forms_id' => $form->getID(),
+            ],
+        ]);
+        foreach ($rows as $row) {
+            $next_index = $row['current_index'] + 1;
+        }
```

Comment:
You will have only one result with a `MAX` query.
```suggestion
        $next_index = $rows->current()['current_index'] + 1;
```